mmv1/products/edgecontainer/NodePool.yaml

```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: "NodePool"
+description: |
+  "A set of Kubernetes nodes in a cluster with common configuration and specification."
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Google Distributed Cloud Edge": "https://cloud.google.com/distributed-cloud/edge/latest/docs"
+  api: "https://cloud.google.com/distributed-cloud/edge/latest/docs/reference/container/rest/v1/projects.locations.clusters.nodePools"
+base_url: "projects/{{project}}/locations/{{location}}/clusters/{{cluster}}/nodePools"
+self_link: 'projects/{{project}}/locations/{{location}}/clusters/{{cluster}}/nodePools/{{name}}'
+create_url: "projects/{{project}}/locations/{{location}}/clusters/{{cluster}}/nodePools?nodePoolId={{name}}"
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+import_format: ["projects/{{project}}/locations/{{location}}/clusters/{{cluster}}/nodePools/{{name}}"]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_node_pool"
+    primary_resource_id: "default"
+    skip_test: true
```

Comment:
Skip the default node-pool test as we only have limited machine resources, and the node-pool creation is tested in edgecontainer_local_control_plane_node_pool_internal to avoid exposing the node location and machine name to public.